linchpin/hooks/__init__.py

```diff
@@ -0,0 +1,200 @@
+"""
+example Pinfile for reference::
+---
+openstack:
+  topology: ex_os_server.yml
+  layout: openshift-3node-cluster.yml
+  hooks:
+    postup:              # sub-state is specified
+        - name: do_something
+          type: shell
+          actions:
+            - samvaran
+        - name: manipulate_inventory
+          type: shell
+          path: /tmp/shellscripts
+          actions:
+            - thisisshell.sh
+        - name: post_up
+          type: ansible
+          actions:
+            - playbook: test_playbook.yaml
+              vars: test_var.yaml
+              extra_vars: { "testvar": "world"}
+    preup:              # sub-state is specified
+        - name: do_something
+          type: shell
+          actions:
+            - echo " this is post up operation Hello hai how r u ?"
+        - name: build_openshift_cluster
+          type: ansible
+          actions:
+            - playbook: test_playbook.yaml
+              vars: test_var.yaml
+              extra_vars: { "testvar": "world"}
+    postdestroy:
+        - name: do_something
+          type: shell
+          actions:
+            - echo " this is post up operation Hello hai how r u ?"
+        - name: postdown_task
+          type: ansible
+          actions:
+            - playbook: test_playbook.yaml
+              vars: test_var.yaml
+              extra_vars: { "testvar": "world"}
+
+"""
+
+import pprint
+import ast
+import sys
+
+from linchpin.hooks.action_managers import ACTION_MANAGERS
+from linchpin.exceptions import ActionManagerError
+
+class ActionBlockRouter(object):
+    """
+    proxy pattern implementation for fetching actionmanagers by name
+
+    """
+    def __init__(self, name, *args, **kwargs):
+
+        self.__implementation = self.__get_implementation(name)(name, *args, **kwargs)
+
+    def __getattr__(self, name):
+
+        return getattr(self.__implementation, name)
+
+    def __get_implementation(self, class_name):
+        """
+        Fetches implementation of lcass from linchpin.hooks.action_managers
+        :param name: action manager class name  
+        """
+
+        action_class = ACTION_MANAGERS.get(class_name, None)
+        if action_class == None:
+            raise ActionManagerError("Action Class %s not found " % (class_name))
+        return action_class
+
+class LinchpinHooks(object):
+
+    def __init__(self, api):
+
+        self.api = api
+        self.api.bind_to_state(self.run_hooks)
+
+    def prepare_ctx_params(self):
+        
+        """
+        prepares few context parameters based on the current target_data
+        that is being set. these parameters are based topology name.
+        """
+        
+        topology = self.api.current_target_data["topology"]
+        name = topology.split("/")[-1].split(".")[-2]
+        inv_file = name+".inventory"
+        res_file = name+".output"
```

Comment:
These two extensions seem like hardcoded values. I know that generally speaking, the names of the files will be called something.inventory and something.output, but might be something different (something.resources maybe?) in the future. My point here, though, is if we change the extension, we have to find this entry in the file while debugging. Can these values be stored in linchpin.conf or some such?